pymeasure/instruments/agilent/agilent8722ES.py

```diff
@@ -179,31 +192,32 @@ def frequencies(self):
         )
 
     @property
-    def data(self):
-        """ Returns the real and imaginary data from the last scan
+    def data_complex(self):
+        """ Returns the complex power from the last scan
         """
         # TODO: Implement binary transfer instead of ASCII
         data = np.loadtxt(
-            BytesIO(self.ask("FORM4;OUTPDATA")),
+            BytesIO(self.ask("FORM4;OUTPDATA").encode()),
             delimiter=',',
             dtype=np.float32
         )
-        return data[:, 0], data[:, 1]
+        data_complex = data[:, 0] + 1j * data[:, 1]
+        return data_complex
 
-    def log_magnitude(self, real, imaginary):
-        """ Returns the magnitude in dB from a real and imaginary
-        number or numpy arrays
+    @property
+    def data_log_magnitude(self):
+        """ Returns the magnitude in dB from the last scan
         """
-        return 20*np.log10(self.magnitude(real, imaginary))
+        return 20*np.log10(self.data_magnitude())
 
-    def magnitude(self, real, imaginary):
-        """ Returns the magnitude from a real and imaginary
-        number or numpy arrays
+    @property
+    def data_magnitude(self):
+        """ Returns the absolute magnitude from the last scan
         """
-        return np.sqrt(real**2 + imaginary**2)
+        return np.sqrt(np.abs(self.data_complex()))
```

Comment:
As `self.data_complex` is a property, I think you don't need to call it as `self.data_complex()` but can just use `self.data_complex`.